lib/mistral_ai_summary_example/settings_dialogs.dart

```diff
@@ -0,0 +1,262 @@
+                value: _maxTokensController.text.isEmpty,
+                onChanged: null,
+                activeTrackColor: Theme.of(context).colorScheme.primary,
+              ),
+            ],
+          ),
+          const SizedBox(height: 24),
+        ],
+      ),
+    );
+  }
+}
+
+class RandomSeedDialog extends StatefulWidget {
+  const RandomSeedDialog({
+    super.key,
+  });
+
+  @override
+  State<RandomSeedDialog> createState() => _RandomSeedDialogState();
+}
+
+class _RandomSeedDialogState extends State<RandomSeedDialog> {
+  final TextEditingController _randomSeedController = TextEditingController();
+
+  @override
+  void initState() {
+    _randomSeedController.text =
+        context.read<SummarySettingsModel>().settings.randomSeed?.toString() ??
+            '';
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _randomSeedController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return CommonSettingsDialog(
+      title: 'Random Seed',
+      description: 'Write a number if you want to set a random seed',
+      onSettingChanged: () {
+        final randomSeed = _randomSeedController.text.isEmpty
+            ? null
+            : int.tryParse(_randomSeedController.text);
+        context.read<SummarySettingsModel>().setRandomSeed(randomSeed);
+        Navigator.of(context).pop();
+      },
+      child: Column(
+        children: [
+          const SizedBox(height: 24),
+          TextField(
+            controller: _randomSeedController,
+            onChanged: (_) => setState(() {}),
+            decoration: InputDecoration(
+              border: const OutlineInputBorder(),
+              suffixIcon: InkWell(
+                borderRadius: BorderRadius.circular(20),
+                onTap: () => setState(_randomSeedController.clear),
+                child: const Icon(Symbols.cancel),
+              ),
+              hintText: 'e.g. 123',
+            ),
+          ),
+          const SizedBox(height: 16),
+          Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: [
+              const Text(
+                'Unset',
+              ),
+              Switch(
+                value: _randomSeedController.text.isEmpty,
+                onChanged: null,
+                activeTrackColor: Theme.of(context).colorScheme.primary,
+              ),
+            ],
+          ),
+          const SizedBox(height: 24),
+        ],
+      ),
+    );
+  }
+}
+
+class CommonSettingsDialog extends StatelessWidget {
+  const CommonSettingsDialog({
+    required this.title,
+    required this.description,
+    required this.child,
+    required this.onSettingChanged,
+    super.key,
+  });
+
+  final String title;
+  final String description;
+  final Widget child;
+  final VoidCallback onSettingChanged;
+
+  @override
+  Widget build(BuildContext context) {
+    return Dialog(
+      child: SizedBox(
+        width: 312,
```

Comment:
Why 312?

Reply:
I took this value from the design on Figma. Also when I tested it on mobile it looked OK.
![image](https://github.com/nomtek/flutter_ai_examples/assets/55763510/3160135b-dd65-4955-9070-28150ef1241e)

Reply:
Interesting. Let's just add a comment there that's a value from Figma design.

Reply:
Resolved in [1c088b2](https://github.com/nomtek/flutter_ai_examples/pull/30/commits/1c088b280cf4ba2497887b3b95f435b78397614f)